transport/tchannel/channel_inbound_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tchannel
+
+import (
+	"testing"
+	"time"
+
+	"go.uber.org/yarpc/api/transport/transporttest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/json"
+)
+
+func TestChannelInboundStartNew(t *testing.T) {
+	tests := []struct {
+		withInbound func(*tchannel.Channel, func(*ChannelInbound))
+	}{
+		{
+			func(ch *tchannel.Channel, f func(*ChannelInbound)) {
+				x, err := NewChannelTransport(WithChannel(ch))
+				require.NoError(t, err)
+
+				i := x.NewInbound()
+				i.SetRouter(new(transporttest.MockRouter))
+				// Can't do Equal because we want to match the pointer, not a
+				// DeepEqual.
+				assert.True(t, ch == i.Channel(), "channel does not match")
+				require.NoError(t, i.Start())
+				defer i.Stop()
+				require.NoError(t, x.Start())
+				defer x.Stop()
+
+				f(i)
+			},
+		},
+		{
+			func(ch *tchannel.Channel, f func(*ChannelInbound)) {
+				x, err := NewChannelTransport(WithChannel(ch))
+				require.NoError(t, err)
+
+				i := x.NewInbound()
+				i.SetRouter(new(transporttest.MockRouter))
+				assert.True(t, ch == i.Channel(), "channel does not match")
+				require.NoError(t, i.Start())
+				defer i.Stop()
+				require.NoError(t, x.Start())
+				defer x.Stop()
+
+				f(i)
+			},
+		},
```

Comment:
Wait... aren't these two test cases the same?

![](https://68.media.tumblr.com/e62214202bfe22d05749a387ff7796e0/tumblr_nm3bw3v9ll1u7qqzwo1_400.gif)